src/main/java/net/pms/network/UPNPHelper.java

```diff
@@ -216,11 +216,13 @@ private static void sendReply(String host, int port, String msg) {
 	public static void sendAlive() {
 		LOGGER.debug("Sending ALIVE...");
 		MulticastSocket multicastSocket = null;
-
+		SocketAddress sa = null;
+		NetworkInterface ni = null;
 		try {
 			multicastSocket = getNewMulticastSocket();
-			InetAddress upnpAddress = getUPNPAddress();
-			multicastSocket.joinGroup(upnpAddress);
+			sa = new InetSocketAddress(getIPv4MulticastAddress(), UPNP_PORT);
+			ni = PMS.get().getServer().getNetworkInterface();
+			multicastSocket.joinGroup(sa, ni);
```

Comment:
I know this is already like this in master, but I wonder if we need to keep joining the group every time - can we just stay joined? There seems to be some CPU overhead associated with sending ALIVE messages and I wonder if that is the reason

Reply:
@SubJunk this was also my question when I made those changes. Here is the change I made for testing the new approach.